src/Terminus.php

```diff
@@ -2,47 +2,90 @@
+    }
+
+    /**
+     * Runs the instantiated Terminus application
+     *
+     * @param string[] $arguments Argv from the command line
+     * @return integer The exiting status code of the application
+     */
+    public function run(array $arguments = [])
+    {
         $config = new Config();
-        parent::__construct('Terminus', $config->get('version'));
+        $status_code = $this->runner->execute($arguments, null, 'Terminus', $config->get('version'));
+        return $status_code;
     }
 
     /**
-     * @inheritdoc
+     * Discovers command classes using CommandFileDiscovery
+     *
+     * @param string[] $options Elements as follow
+     *        string path      The full path to the directory to search for commands
+     *        string namespace The full namespace associated with given the command directory
+     * @return string[] A hash wherein the id is the file name and the value the class of a command file
      */
-    protected function getDefaultHelperSet()
+    private function discoverCommands(array $options = ['path' => null, 'namespace' => null,])
     {
-        $helper_set = parent::getDefaultHelperSet();
-        //$helper_set->set(new Pantheon\Terminus\Helpers\FileHelper());
-        return $helper_set;
+        $discovery = new CommandFileDiscovery();
+        $discovery->setSearchPattern('*Command.php')->setSearchLocations(['.']);
+        $commands = $this->fixNames($discovery->discover($options['path'], $options['namespace']));
+        return $commands;
     }
 
     /**
-     * @inheritdoc
+     * Adds command files to the application
+     *
+     * @return void
      */
-    protected function getDefaultInputDefinition()
+    private function setRunner(Runner $runner)
     {
-        $definition = parent::getDefaultInputDefinition();
-        $definition->addOptions(
-            [
-            new InputOption(
-                '--yes',
-                '-y',
-                InputOption::VALUE_NONE,
-                'Answer yes to all prompts'
-            ),
-            ]
+        $this->runner = $runner;
+    }
+
+    /**
+     * Fixes the extraneous '\.' and '/.' currently inserted by the discovery and removes abstract classes
+     *
+     * @param string[] $command_files A hash of command file names and paths
+     * @return string[] The param array, fixed
+     */
+    private function fixNames($command_files) {
+        $commands = array_filter(
+          array_combine(
+            array_map(
+                function ($file_name) {return str_replace('/./', '/', $file_name);},
+                array_keys($command_files)
+              ),
+              array_map(
+                function ($class_name) {return str_replace('\\.\\', '\\', $class_name);},
+                array_values($command_files)
+            )
+          ),
+          function ($class_name) {
+              $reflection_class = new \ReflectionClass($class_name);
+              return !$reflection_class->isAbstract();
```

Comment:
This is okay, but if you just put your abstract classes in a different namespace, then you wouldn't need to filter them out.